### PR TITLE
cli: clean-up the handling of default flags

### PR DESCRIPTION
**Pull request by knz**
See individual commits.

### CODE REVIEW

File: pkg/cli/demo.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ func runDemo(cmd *cobra.Command, gen workload.Generator) (err error) {
 		return err
 	}
 
-	loc, err := geos.EnsureInit(geos.EnsureInitErrorDisplayPrivate, demoCtx.geoLibsDir)
+	loc, err := geos.EnsureInit(geos.EnsureInitErrorDisplayPrivate, startCtx.geoLibsDir)
```
Comment (otan) [inline pkg/cli/demo.go:267]: woops, thanks!

File: pkg/cli/flags.go [hunk 2]
```diff
@@ -663,24 +663,32 @@ func init() {
...
+		// ./cockroach demo movr --nodes=3.
+		f := demoCmd.PersistentFlags()
+
+		intFlag(f, &demoCtx.nodes, cliflags.DemoNodes)
+		boolFlag(f, &demoCtx.runWorkload, cliflags.RunDemoWorkload)
+		varFlag(f, &demoCtx.localities, cliflags.DemoNodeLocality)
+		boolFlag(f, &demoCtx.geoPartitionedReplicas, cliflags.DemoGeoPartitionedReplicas)
+		varFlag(f, demoNodeSQLMemSizeValue, cliflags.DemoNodeSQLMemSize)
+		varFlag(f, demoNodeCacheSizeValue, cliflags.DemoNodeCacheSize)
+		boolFlag(f, &demoCtx.insecure, cliflags.ClientInsecure)
```
Comment (otan) [inline pkg/cli/flags.go:678]: should we doing the same thing with the insecure flag?

**Review by otan — COMMENTED** [review]
LGTM, one question pending.

File: pkg/cli/flags.go (hunk 2 above)
Comment (knz) [reply pkg/cli/flags.go:678]: unfortunatey `start` and `demo` use different defauts

**Comment by knz** [comment]
TFYR

bors r=otan

**Pull request merged**